apps/demo/src/routes/logcat.tsx

```diff
@@ -0,0 +1,230 @@
+import { IconButton, SearchBox, Stack, StackItem } from '@fluentui/react';
+import { AdbShell } from '@yume-chan/adb';
+import { encodeUtf8 } from '@yume-chan/adb-backend-webusb';
+import { AutoDisposable } from '@yume-chan/event';
+import { CSSProperties, useCallback, useContext, useEffect, useRef, useState } from 'react';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+import { SearchAddon } from 'xterm-addon-search';
+import 'xterm/css/xterm.css';
+import { ErrorDialogContext } from '../components/error-dialog';
+import { ResizeObserver, withDisplayName } from '../utils';
+import { RouteProps } from './type';
+
+const ResizeObserverStyle: CSSProperties = {
+    width: '100%',
+    height: '100%',
+};
+
+const UpIconProps = { iconName: 'ChevronUp' };
+const DownIconProps = { iconName: 'ChevronDown' };
+
+class AdbTerminal extends AutoDisposable {
```

Comment:
Maybe move the class to a separete file to reuse it. Or even extract the whole component so it can be used like `<XtermTerminal socket={socket} />`